lib/run/index.js

```diff
@@ -82,7 +82,7 @@ module.exports = function (options, callback) {
         runner.run(options.collection, {
             stopOnFailure: options.bail, // LOL, you just got trolled ¯\_(ツ)_/¯
             abortOnFailure: options.abortOnFailure,
-            iterationCount: options.iterationCount,
+            iterationCount: options.iterationCount || _.get(options, 'iterationData.length'),
```

Comment:
@czardoz just a thought - if runtime already handles this, we shouldn't even try parsing anything from iteration data.